weave-js/src/components/PagePanelComponents/Home/HomePreviewSidebar.tsx

```diff
@@ -149,10 +172,96 @@ export const HomeExpressionPreviewParts: React.FC<{
           {/* </Unclickable> */}
         </LayoutElements.Block>
       </LayoutElements.VBlock>
+      {generators.loading || refinedExpression.loading || isGenerating ? (
+        <LayoutElements.VStack
+          style={{
+            alignItems: 'center',
+            justifyContent: 'center',
+            gap: '16px',
+            color: '#8e949e',
+          }}>
+          <WeaveAnimatedLoader style={{height: '64px', width: '64px'}} />
+        </LayoutElements.VStack>
+      ) : (
+        generators.result.length > 0 && (
+          <LayoutElements.VBlock style={{gap: '8px'}}>
+            <span style={{color: '#2B3038', fontWeight: 600}}>
+              Available Templates
+            </span>
+            <LayoutElements.VStack
+              style={{
+                gap: '8px',
+              }}>
+              {generators.result.map(template => (
+                <DashboardTemplate
+                  key={template.op_name}
+                  title={template.display_name}
+                  subtitle={template.description}
+                  onClick={() => {
+                    setIsGenerating(true);
+                    makeBoardFromNode(
+                      template.op_name,
+                      refinedExpression.result as any,
+                      newDashExpr => {
+                        navigateToExpression(newDashExpr);
+                        setIsGenerating(false);
+                      }
+                    );
+                  }}
+                />
+              ))}
+            </LayoutElements.VStack>
+          </LayoutElements.VBlock>
+        )
+      )}
     </LayoutElements.VStack>
   );
 };
 
+const DashboardTemplate: React.FC<{
```

Comment:
I'm in favor of not declaring things with `React.FC` (reasons described in 
https://www.harrymt.com/blog/2020/05/20/react-typescript-react-fc.html) - but I don't think we've officially standardized on any style guidance yet.

Reply:
we might need to refactor the whole codebase on this one